markdown-mode.el

```diff
@@ -3509,17 +3509,30 @@ SEQ may be an atom or a sequence."
                                    `(display ,display-string))))))))
     t))
 
+(defun markdown--fontify-hrs-view-mode (hr-char)
+  (if (and hr-char (display-supports-face-attributes-p '(:underline t)))
```

Comment:
Maybe, instead of checking for `:underline`, you could check for `:extend`.  I don't know when Emacs started supporting the latter (maybe someone does), but I'm fairly sure most a good number of platforms and Emacs versions support :underline.

The other reason I say this is because I _think_ that with `:extend t` you can make a "bar" out of any text decoration feature, not just `:underline`.